woocommerce/woocommerce-ios
# Product Subscriptions: Enable creating and editing subscription types

Closes: #11153 


## What

This PR adds changes to support creating product subscriptions. Some small issues related to product creation are also added to polish the feature further.

## How
- Added an extension for `ProductSubscription` to create an empty subscription for product creation.
- Updated `ProductFactory` to include subscription types for product creation.
- Updated `ProductFormActionsFactory` to enable updating product type for simple subscription products.
- Updated `ProductTypeBottomSheetListSelectorCommand` to include subscription types when changing product types.
- Other issues:
  - Fixed the truncated labels on the product type bottom sheet by only triggering layout for the table view after the view is added to a window.
  - Restored the missing dimmed background for the product creation bottom sheet.

## Testing instructions

- Enable the feature flag `subscriptionProducts` to be able to add or edit subscription products.
- Create a simple subscription product for your test site on wp-admin if you haven't already.
- Build the app, log in to your test store, and navigate to the Products tab.
- Tap "+" to add a new product. Notice that there is a dimmed background behind the bottom sheet.
- Select the manual option. Notice that no description label is truncated.
- Select Simple subscription product.
- Notice that the app doesn't crash, and the product form is displayed instead. 
- Tap on Price, notice that there are fields for subscription products.
- Navigate back, tap Product type > select a Simple Physical Product.
- Tap on Price, notice that there are no fields for subscription products.

The app should not crash when you tap Variable Subscription Product for creation, but the form will be updated in a separate task.

Please also feel free to test again with the feature flag `subscriptionProducts` disabled. When trying to change a product type, there should be no option for subscription products.

## Screenshots

Adding a simple subscription product:

https://github.com/woocommerce/woocommerce-ios/assets/5533851/03d4b7ba-7fe4-4ec0-bfdc-2b58550b062f

Changing product type from/to a simple subscription product:


https://github.com/woocommerce/woocommerce-ios/assets/5533851/81b21ee1-dce7-469e-9bb9-c4489d5f96eb


---
